liquibase-core/src/main/java/liquibase/diff/compare/DatabaseObjectComparatorChain.java

```diff
@@ -82,11 +82,11 @@ public ObjectDifferences findDifferences(DatabaseObject object1, DatabaseObject
         if ((object1 == null) && (object2 == null)) {
             return new ObjectDifferences(compareControl);
         }
-        if ((object1 == null) && (object2 != null)) {
+        if (object1 == null) {
             return new ObjectDifferences(compareControl).addDifference("Reference value was null", "this", null, null);
```

Comment:
It's not directly related with the changes made here, but I'm seeing the only difference between the two `ObjectDifferences` object is the message set. I think this should be something like:
```
return new ObjectDifferences(compareControl).addDifference("Reference value was null", "this", null, object2);```
and:
```
return new ObjectDifferences(compareControl).addDifference("Reference value was null", "this", object1, null);```
```

Reply:
'object2 != null))' this is true always when reach the conditions.